### PR TITLE
[Merged by Bors] - feat: expand API for `CFC.posPart`

### PR DESCRIPTION
**Pull request by j-loreaux**
---

- [ ] depends on: #19216 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/PosPart.lean [hunk 1]
```diff
@@ -94,22 +140,46 @@ lemma negPart_nonneg (a : A) :
...
+  have ha := hac.symm ▸ (posPart_nonneg a).isSelfAdjoint.sub hc.isSelfAdjoint
+  nth_rw 1 [← posPart_sub_negPart a] at hac
+  simpa using hac
+
+lemma le_posPart {a : A} (ha : IsSelfAdjoint a) : a ≤ a⁺ := by
+  simpa [posPart_sub_negPart a] using sub_le_self a⁺ (negPart_nonneg a)
+
+lemma neg_negPart_le {a : A} (ha : IsSelfAdjoint a) : -a⁻ ≤ a := by
+  simpa only [posPart_sub_negPart a, ← sub_eq_add_neg]
+    using le_add_of_nonneg_left (a := -a⁻) (posPart_nonneg a)
```
Comment (dupuisf) [inline Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/PosPart.lean:160]:
```suggestion
lemma le_posPart {a : A} (ha : IsSelfAdjoint a := by cfc_tac) : a ≤ a⁺ := by
  simpa [posPart_sub_negPart a] using sub_le_self a⁺ (negPart_nonneg a)

lemma neg_negPart_le {a : A} (ha : IsSelfAdjoint a := by cfc_tac) : -a⁻ ≤ a := by
  simpa only [posPart_sub_negPart a, ← sub_eq_add_neg]
    using le_add_of_nonneg_left (a := -a⁻) (posPart_nonneg a)
```

File: Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/PosPart.lean [hunk 2]
```diff
@@ -94,22 +140,46 @@ lemma negPart_nonneg (a : A) :
...
+lemma negPart_eq_zero_iff (a : A) (ha : IsSelfAdjoint a := by cfc_tac) :
     a⁻ = 0 ↔ 0 ≤ a := by
-  rw [← eq_posPart_iff]
+  rw [← posPart_eq_self, eq_comm (b := a)]
   nth_rw 2 [← posPart_sub_negPart a]
   simp
 
-lemma eq_negPart_iff (a : A) : a = -a⁻ ↔ a ≤ 0 := by
+lemma negPart_eq_neg (a : A) : a⁻ = -a ↔ a ≤ 0 := by
+  rw [← neg_inj, neg_neg, eq_comm]
```
Comment (dupuisf) [inline Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/PosPart.lean:182]:
```suggestion
lemma negPart_eq_neg_iff (a : A) : a⁻ = -a ↔ a ≤ 0 := by
  rw [← neg_inj, neg_neg, eq_comm]
```
This feels a bit more in line with the rest, no?

**Review by dupuisf — COMMENTED** [review]
Looks good, thanks!

bors d+

File: Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/PosPart.lean (hunk 2 above)
Comment (j-loreaux) [reply Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/PosPart.lean:182]: no, I updated these names to match [posPart_eq_self](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Group/PosPart.html#posPart_eq_self) and [negPart_eq_neg](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Group/PosPart.html#negPart_eq_neg)

**Comment by j-loreaux** [comment]
bors merge

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**